sdk/keyvault/keyvault-certificates/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -93,35 +128,37 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
       throw new Error("The resource address for authorization must use the 'https' protocol.");
     }
 
-    let originalBody = webResource.body;
+    const originalBody = webResource.body;
 
-    if (this.challenge == undefined) {
+    if (this.challengeCache.challenge == undefined) {
```

Comment:
This feels a little surprising to not be a cache lookup here.

Reply:
There's only one challenge we're caching. The reason we're "caching" it is because the class gets re-generated on every request. There's no multitude of challenges we will be receiving in these pipelines, as far as I'm aware.